packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -15,6 +17,27 @@ export interface ContactEntry {
   importTime?: number;
 }
 
+type EtherscanSupportedHexChainId =
+  | '0x1'
+  | '0x5'
+  | '0x38'
+  | '0x61'
+  | '0xa'
+  | '0x1a4'
+  | '0x89'
+  | '0x13881'
+  | '0xa86a'
+  | '0xa869'
+  | '0xfa'
+  | '0xfa2'
+  | '0xaa36a7'
+  | '0xe704'
+  | '0xe708'
+  | '0x504'
+  | '0x507'
+  | '0x505'
+  | '0x64';
+
```

Comment:
To make this logic less brittle and easier to update, we could use `ETHERSCAN_SUPPORTED_CHAIN_IDS` as a single source of truth.

```suggestion

export type EtherscanSupportedChains =
  keyof typeof ETHERSCAN_SUPPORTED_CHAIN_IDS;

export type EtherscanSupportedHexChainId =
  (typeof ETHERSCAN_SUPPORTED_CHAIN_IDS)[EtherscanSupportedChains];
```

Reply:
Will add this changes asap, thanks for the review @MajorLift